### PR TITLE
Jean/update heterogeneity plots styling + fix Fed-ISIC umap reproducibility

### PR DESCRIPTION
**Pull request by jeandut**
This PR allows to produce the heterogeneity plots forming the figure of the article that can be seen in openreview.
Reviewers should focus on kits19 plotting, which has significantly changed and ISIC other changes are normally cosmetic.

### CODE REVIEW

File: flamby/datasets/fed_isic2019/heterogeneity_pic.py [hunk 1]
```diff
@@ -43,14 +47,21 @@ def forward(self, image):
...
     )
+    parser.add_argument(
+        "--seed",
+        type=int,
+        default=42,
+        help="The seed for the UMPA and dataloading",
+    )
     args = parser.parse_args()
+    np.random.seed(args.seed)
+    torch.manual_seed(42)
```
Comment (jeandut) [inline flamby/datasets/fed_isic2019/heterogeneity_pic.py:61]:
```suggestion
    torch.manual_seed(args.seed)
    random.seed(args.seed)
```

File: flamby/datasets/fed_isic2019/heterogeneity_pic.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import argparse
 import os
```
Comment (jeandut) [inline flamby/datasets/fed_isic2019/heterogeneity_pic.py:2]:
```suggestion
import os
import random
```

File: flamby/datasets/fed_heart_disease/heterogeneity_plot.py [hunk 3]
```diff
@@ -3,8 +3,9 @@
 import seaborn as sns
 
 from flamby.datasets.fed_heart_disease import FedHeartDisease
+from flamby.utils import seaborn_styling
 
-sns.set_theme()
+seaborn_styling(legend_fontsize=10, labelsize=12)
```
Comment (jeandut) [inline flamby/datasets/fed_heart_disease/heterogeneity_plot.py:8]:
```suggestion
seaborn_styling(legend_fontsize=50, labelsize=12)
```

File: README.md [hunk 4]
```diff
@@ -208,6 +208,7 @@ cd flamby/datasets/fed_lidc_idri
 python lidc_heterogeneity_plot.py
 ```
 - [Fed-ISIC2019](./flamby/datasets/fed_isic2019/README.md)
+**You need to first deactivate color constancy normalization when preprocessing the dataset (cc=False in resize_images.py)**
 ```
```
Comment (regloeb) [inline README.md:212]: It is not clear in the readme that running heterogeneity pics requires preprocessing of the datasets.

**Review by regloeb — COMMENTED** [review]
Not sure my comments are really helpful as most of the changes would require running the code extensively for each heterogeneity pic ...

File: README.md [hunk 5]
```diff
@@ -208,6 +208,7 @@ cd flamby/datasets/fed_lidc_idri
 python lidc_heterogeneity_plot.py
 ```
 - [Fed-ISIC2019](./flamby/datasets/fed_isic2019/README.md)
+**You need to first deactivate color constancy normalization when preprocessing the dataset (cc=False in resize_images.py)**
```
Comment (jeandut) [inline README.md:211]:
```suggestion
**In order to exactly reproduce the plot in the article, one needs to first deactivate color constancy normalization when preprocessing the dataset (change `cc` to `False` in `resize_images.py`) when following download and preprocessing instructions [here](./flamby/datasets/fed_isic2019/README.md).  Hence one might require to download the dataset a second time, if it was already downloaded, and to potentially update `dataset_location.yaml` files accordingly.**
```

File: README.md [hunk 6]
```diff
@@ -227,7 +228,16 @@ python kits19_heterogenity_plot.py
 cd flamby/datasets/fed_heart_disease
 python heterogeneity_plot.py
 ```
-
+- [Fed-Camelyon16](./flamby/datasets/fed_camelyon16/README.md)
+First concatenate as many matter image patches from Hospital 0 and Hospital 1 as can be fit in
```
Comment (jeandut) [inline README.md:232]:
```suggestion
First concatenate as many 224x224 image patches extracted from regions on the slides containing matter from Hospital 0 and Hospital 1 (see what is done in the [tiling script](./flamby/datasets/fed_camelyon16/dataset_creation_scripts/tiling_slides.py) as can be fit in
```

File: flamby/datasets/fed_kits19/dataset_creation_scripts/kits19_heterogenity_plot.py [hunk 7]
```diff
@@ -25,75 +28,81 @@ def add_args(parser):
...
-                array = np.array(nib.load(image_file).dataobj)
-                plot_histogram(ax, array, color=colours[silo_count])
-                iteration += 1
+                array = np.array(nib.load(image_file).dataobj).astype("float32")
+                # We have finite RAM unfortunately
+                if silo_array.shape[0] > 1e9:
+                    break
+                silo_array = np.concatenate((silo_array, array.ravel()))
+            plot_histogram(ax, silo_array, color=colours[silo_count])
             silo_count += 1
```
Comment (regloeb) [inline flamby/datasets/fed_kits19/dataset_creation_scripts/kits19_heterogenity_plot.py:89]: looks reasonable to me

**Pull request merged**